### ISSUE TITLE
fix identification of source blocks

### ISSUE THREAD

**Issue by jaredh159** [jaredh159/asciidork#4]
we currently fail to recognize

```adoc
[#hello,ruby]
----
require 'sinatra'
----
```

because i guess i misunderstood the identification of source blocks, i thought it needed to has some string value in the _second_ position, based on [this](https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-blocks/), but [later examples](https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-highlighter/#apply-source-highlighting) show the `#hello` inference as well.

**Comment by SteveAlexander** [comment]
If it helps, some illuminating asciidoc tests are here, demonstrating a broad range of what variations asciidoc will process:

https://github.com/asciidoctor/asciidoctor/blob/main/test/syntax_highlighter_test.rb

In all cases, the language identifier is in the second position. What happens in the other positions looks like it can get complicated!

**Comment by jaredh159** [comment]
yeah, thanks that is helpful. i frequently work through the tests cases, but hadn't looked at that file yet. good to know 👍
